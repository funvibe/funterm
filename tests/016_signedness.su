lua.data = <<255:8, 127:8, 128:8>>

match lua.data {
    <<a:8/unsigned, b:8/unsigned, c:8/unsigned>> -> {
        lua.print("Unsigned:", a, b, c)
    }
}

match lua.data {
    <<a:8/signed, b:8/signed, c:8/signed>> -> {
        lua.print("Signed:", a, b, c)
    }
}

# Отрицательные числа
lua.negative = <<-1:8/signed, -128:8/signed, 127:8/signed>>
match lua.negative {
    <<a:8/signed, b:8/signed, c:8/signed>> -> {
        lua.print("Negative signed:", a, b, c)
    }
}