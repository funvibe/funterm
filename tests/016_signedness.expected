Unsigned: 255 127 128
Signed: -1 127 -128
Negative signed: -1 -128 127